tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2103,16 +2112,16 @@ static bool CombineBasicTypes(ArBasicKind LeftKind,
                               ArBasicKind RightKind,
                               _Out_ ArBasicKind* pOutKind)
 {
-  if ((LeftKind < 0 || LeftKind >= AR_BASIC_COUNT) ||
-    (RightKind < 0 || RightKind >= AR_BASIC_COUNT)) {
-    return false;
-  }
-
   if (LeftKind == RightKind) {
     *pOutKind = LeftKind;
     return true;
   }
 
+  if ((LeftKind < 0 || LeftKind >= AR_BASIC_COUNT) ||
```

Comment:
Moving this block down seems a little odd. Should we allow void or out of range values if they match?

Reply:
Greater than BASIC isn't out of range, it just means that it's not a basic type.  This allows a non-basic type. I could maintain the void and add a true out of range check here though.

Reply:
Ah, makes sense.